src/components/HeatMap/HeatMap.js

```diff
@@ -1,16 +1,71 @@
-import React from 'react';
-import { Link } from 'react-router-dom';
-import table from './table.png';
-import config from '../../config';
-import { Wrapper, Image } from './HeatMap.styles';
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import Row from './Row';
+import Header from './Header';
+import Footer from './Footer';
+import { Wrapper } from './HeatMap.styles';
 
 
-export default function HeatMap() {
+const dayLabels = [
+  'Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday',
+];
+
+export default function HeatMap({ data }) {
+  const [tableData, setTableData] = useState([]);
+  const [activeCell, setActiveCell] = useState([]);
+  const [timezone, setTimezone] = useState('');
+  const [timezoneDifference, setTimezoneDifference] = useState(0);
+
+
+  useEffect(() => {
+    const clientTimezone = Intl.DateTimeFormat().resolvedOptions().timeZone;
+    const clientTimezoneDifference = (new Date().getTimezoneOffset()) * 60;
+    setTimezone(clientTimezone);
+    setTimezoneDifference(clientTimezoneDifference);
+  }, []);
+
+  useEffect(() => {
+    const buildTable = () => {
+      const orderPosts = new Array(7).fill([]).map(() => new Array(24).fill([]));
+
+      data.forEach((item) => {
+        const clientDate = (item.data.created_utc + timezoneDifference) * 1000;
+        const date = new Date(clientDate);
+        const day = Number(date.getDay());
+        const time = Number(date.toLocaleTimeString('de-De').slice(0, 2));
+        orderPosts[day][time] = [...orderPosts[day][time].concat(item.data)];
+      });
+      setTableData(orderPosts);
+    };
+
+    buildTable();
+  }, [data, timezoneDifference]);
+
+  const toggleActiveCell = (activeRow, activeColumn) => {
```

Comment:
Yeah, the naming is a bit annoying here since `setActiveCell` is already taken. `toggle` would mean that the active cell would be set to `true` or `false` depending on the current state. So this is not a toggle in a strict sense.

You could simply call it `onClickCell` as an alternative